### PR TITLE
keyValue matcher, bump versions

### PR DESCRIPTION
**Pull request by waschmittel**

### CODE REVIEW

File: src/main/java/de/dm/infrastructure/logcapture/ExpectedKeyValueLogstashDelegate.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import net.logstash.logback.marker.SingleFieldAppendingMarker;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+
+// this is a delegate accessing net.logstash so that ExpectedKeyValue can fail gracefully
```
Comment (SchlottiP) [inline src/main/java/de/dm/infrastructure/logcapture/ExpectedKeyValueLogstashDelegate.java:12]: Why did you remove logstash in the build but than verify that it exists?

Comment (waschmittel) [reply src/main/java/de/dm/infrastructure/logcapture/ExpectedKeyValueLogstashDelegate.java:12]: Logstash is a build-time dependency so it can be used for assertions **if** it is in the classpath.

However, it should not be a dependency. This is why I check at runtime if it is present.

This is also why this package-private delegate exists: It will never be loaded if logstash ist not in the classpath to avoid ClassNotFoundException

File: src/main/java/de/dm/infrastructure/logcapture/LogEventMatcher.java [hunk 2]
```diff
@@ -27,14 +27,14 @@
...
+    String getMatcherTypeDescription();
 
     /**
-     * returns an error message describing why a logged event does match, even of it shouldn't.
+     * returns an error message describing the concrete matcher, including as much of its expectations as possible
      *
      * @return matched aspect
      */
-    String getMatchingErrorMessage();
+    String getMatcherDetailDescription();
```
[the remark this answers is not in the document]
Comment (waschmittel) [reply src/main/java/de/dm/infrastructure/logcapture/LogEventMatcher.java:37]: In short: Because it was used ambiguously and not only for the error case. Also, the description is easier to formulate if it only needs to describe the matcher and does not need to concern itself with the context in which it is used.

Let's discuss this in detail in person.
